### PR TITLE
Create the Asset Database

### PR DESCRIPTION
**Pull request by FreneticScribbler**
PR opened for Heroku deployment purposes, it is not quite ready yet!

### CODE REVIEW

**Comment by FreneticScribbler** [comment]
*whistles*

**Comment by FreneticScribbler** [comment]
Stg y'all are fucking psychic keyboard ninjas...

Where do I acquire this power?

**Comment by FreneticScribbler** [comment]
Hmm, we probably don't need to have Travis build both the PR and the commit..

File: RIGS/migrations/0035_auto_20191008_2148.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.0.13 on 2019-10-08 20:48
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```
Comment (davidtaylorhq) [inline RIGS/migrations/0035_auto_20191008_2148.py]: Why does this file exist in the PR? There is already a similar migration in `master`

File: assets/api.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+from rest_framework import serializers, viewsets, status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.generics import get_object_or_404
+from django.db.models import Min
+from django.core.mail import EmailMessage
+from django.template.loader import get_template
+from django.template import Context, RequestContext
+import datetime
```
Comment (davidtaylorhq) [inline assets/api.py]: This is a lot of imports for a small file. I suspect they were copy/pasted, and most can now be removed?

File: assets/management/commands/deleteSampleData.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        from django.conf import settings
+
+        if not (settings.DEBUG):
+            raise CommandError('You cannot run this command in production')
+
+        # self.delete_categories()
+        # self.create_statuses()
+        # self.create_suppliers()
+        # self.create_collections()
+        # self.create_assets()
```
Comment (davidtaylorhq) [inline assets/management/commands/deleteSampleData.py]: Are these commented bits required, or can they be tidied up?

File: requirements.txt [hunk 4]
```diff
@@ -4,14 +4,17 @@ diff-match-patch==20121119
...
+django-filter==2.0.0
+django-widget-tweaks==1.4.3
 django-debug-toolbar==1.9.1
 django-ical==1.4
 django-recaptcha==1.4.0
 django-registration-redux==2.4
 django-reversion==2.0.13
 django-toolbelt==0.0.1
+django-polymorphic==2.1.2
+djangorestframework==3.9.0
```
Comment (davidtaylorhq) [inline requirements.txt]: Is this actually being used anywhere?

File: requirements.txt [hunk 5]
```diff
@@ -4,14 +4,17 @@ diff-match-patch==20121119
...
 Django==2.0.13
+django-filter==2.0.0
+django-widget-tweaks==1.4.3
 django-debug-toolbar==1.9.1
 django-ical==1.4
 django-recaptcha==1.4.0
 django-registration-redux==2.4
 django-reversion==2.0.13
 django-toolbelt==0.0.1
+django-polymorphic==2.1.2
```
Comment (davidtaylorhq) [inline requirements.txt]: Is this being used anywhere? I see it in migrations, but it looks like the fields were later removed

**Review by davidtaylorhq — CHANGES_REQUESTED** [review]
Skimmed through it and left a few comments. In general it looks good to me. Two general suggestions:

1. Can we get shrink down the number of migration files? They've clearly grown as development has happened, but it's quite messy now. You should be able to simply delete them all, then `makemigrations` again to generate a single file. That might break your local databases, but you can just throw them away and generate sample data again

2. There are no tests. Ultimately it's up to you, but I would recommend adding at least a few for the basic functionality. They have saved us a few times with RIGS!

**Comment by davidtaylorhq** [comment]
One more thing - this branch needs re-basing onto master, or have master merged into it. The latter may be easiest.

File: RIGS/migrations/0035_auto_20191008_2148.py (hunk 1 above)
Comment (FreneticScribbler) [reply RIGS/migrations/0035_auto_20191008_2148.py]: makemigrations on the asset branch makes it for whatever reason, investigating.

File: requirements.txt (hunk 5 above)
Comment (mattysmith22) [reply requirements.txt]: This was explored initially to handle the whole "cables are a subset of assets" issue. Has been removed in recent commit

File: requirements.txt (hunk 4 above)
Comment (mattysmith22) [reply requirements.txt]: They are not, djangorestframework is left over from PyAssets, I removed all dependencies for it (that were unused) and removed the dependency. Both are now removed as dependencies

File: assets/management/commands/deleteSampleData.py (hunk 3 above)
Comment (mattysmith22) [reply assets/management/commands/deleteSampleData.py]: I have removed them, again leftover by PyAssets

File: assets/api.py (hunk 2 above)
Comment (mattysmith22) [reply assets/api.py]: They certainly were! `api.py` was not even used, again a leftover. Also sent me looking for more unused imports, they have been also removed

File: RIGS/migrations/0035_auto_20191008_2148.py (hunk 1 above)
Comment (mattysmith22) [reply RIGS/migrations/0035_auto_20191008_2148.py]: https://github.com/nottinghamtec/PyRIGS/commit/1a49bb50e5c1a0625d67f33e0ce7464820441f3f contains a change to a model, after which we branched off to assets. Meant that when `makemigrations` was next run, it caused it to be created on next run.

No worries, will have to make sure it is removed after merge!

**Pull request merged**